### PR TITLE
Install tar and splunk on WEC instance

### LINKED ISSUE #3: Install splunk on WEC in provisioning script^
- [x] Get splunk MSI 
- [x] Put it somewhere sensible 
- [x] Add a ps script to install it

### LINKED ISSUE #6: Install tar binary from windows 2019 ^^^^^
The current CO windows domain in Windows Server 2016. They plan to migrate to 2019 when they move to AWS. 

There's no tar utility in 2016. The suggested way to get one is to get the tar.exe and associated dll from a 2019 machine and stick them somewhere other than system32 and add that to the path. 

- [x] Stick the tar binary in the repo
- [x] Unpack it into c:\Program Files\WinTar 
- [x] Add to the path in the powershell profile
- [x] Run the script as part of the remote-exec for the WEC instance.

### PR DESCRIPTION
**Pull request by danjoneslf**
- [x] Add an exec role which is trusted to assume the forwarder role in splunk-apps/forwarders 
- [x] TF in a separate statefile to avoid breaking trusts on destroy
- [x] Add an install_packages powershell script to the wec instance which: 
   - [x] Assumes the role 
   - [x] Downloads the /packages/* folder from S3 
   - [x] Compares the file hashes of the downloads 
   - [x] Runs the unpack and add to path for tar 
   - [x] Runs the msi unattended for Splunk  
 - [x] Validate that tar and Splunk are both installed on the WEC instance after TF apply

### CODE REVIEW

File: terraform/deployments/103495720024_exec_roles/variables.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+variable "environment" {
+  type        = string
+  description = "One of [ production | staging | demo ]"
+  default     = "staging"
+}
+
+variable "splunk_config_bucket" {
+  description = "Name of the target bucket for retriving config."
```
Comment (alice-carr) [inline terraform/deployments/103495720024_exec_roles/variables.tf]: very nitpicky but there's a typo in retrieving

File: terraform/modules/windows-test-domain/scripts/WEC/install_packages.ps1 [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          "$PSScriptRoot\packages\$LogFile"
+      )
+      Start-Process "msiexec.exe" -ArgumentList $MSIArguments -Wait -NoNewWindow
+    }
+  }
+}
+
+#Write-Host "Cleaning up"
+#Remove-Item -Recurse -Force $PSScriptRoot/packages
+
```
Comment (alice-carr) [inline terraform/modules/windows-test-domain/scripts/WEC/install_packages.ps1:144]: worth removing these commented out lines?

**Review by alice-carr — DISMISSED** [review]
All looks good to me. I left a couple of minor comments which are fine if you choose to ignore

File: terraform/deployments/103495720024_exec_roles/variables.tf (hunk 1 above)
Comment (danjoneslf) [reply terraform/deployments/103495720024_exec_roles/variables.tf]: Thank you - I hate typos.

File: terraform/modules/windows-test-domain/scripts/WEC/install_packages.ps1 (hunk 2 above)
Comment (danjoneslf) [reply terraform/modules/windows-test-domain/scripts/WEC/install_packages.ps1:144]: Ooh yeah - and I need to switch back to the main branch before this is merged.

**Review by alice-carr — APPROVED** [review]

**Pull request merged**